# **Tesla Car Sales Data Analysis Project**

# **Introduction**
The purpose of this analysis is to gain valuable insights and patterns within Tesla car sales data. By thoroughly examining sales data across various regions, products, and time frames, we aim to identify top-performing products, high-performing regions, and the most profitable time periods. These findings will help optimize our sales strategies and enhance overall performance to drive substantial revenue growth.

# **Data Description**

The dataset used in this analysis was obtained from the Kaggle website and contains 85,592 rows and 7 columns. The columns include Model, Period, Country, Purchase_type, Version, Price, and Gross_Profit, providing comprehensive information about the cars sold by the company.

**Data Dictionary:**

The dataset consists of 7 columns with their corresponding data types:

| Column Name                | Column Datatype |
| :------------------------- | :-------------: |
| Model                      |  VARCHAR(50)    |
| Period                     |     DATE        |
| Country                    |  VARCHAR(50)    |
| Purchase_type              |  VARCHAR(50)    |
| Version                    |  VARCHAR(50)    |
| Price                      |  DECIMAL(10,3)  |
| Gross_Profit               |  DECIMAL(10,3)  |


# **Methodology**

1. **Data Collection:**
  - The dataset used for this analysis was sourced from the Kaggle website.

3. **Data Understanding:**

 - Explored the dataset's structure, available columns, data types, and the data dictionary to understand its contents and objectives.

3. **Data Assessment:** 
  - Thorough data quality assessment was conducted, addressing missing values, outliers, inconsistent formatting, and data entry errors.

4. **Data Cleaning and Data Validation:**

 Several tasks were performed to ensure the data was in a usable and reliable format. Here are some key steps I took:
   - Cleaned the data in Excel and adjusted the date format.
   - Added constraints in Microsoft SQL Server, like a `Primary Key` for uniqueness and data integrity.
   - Chose appropriate `data types` for accurate storage and manipulation of data.
   - Enforced `NOT NULL` constraints on relevant columns to ensure data completeness and prevent the insertion of null values.
 
 5. **Feature Engineering**
   - A new column called `month` was created to extract the corresponding month of each transaction, aiding in identifying months with the highest sales and profitability.

6. **SQL Query types used**
- ALTET TABLE
- ALTER COLUMN
- ADD
- UPDATE
- DATENAME()
- Aggregate Functions
- MIN/MAX
- SUM
- COUNT(DISTINCT)
- AVG
- ROW_NUMBER()
- ROUND
- WHERE BETWEEN
- Subqueries 
- STDEV
- GROUP BY
- ORDER BY 

# **Exploratory Data Analysis (EDA)**

**Throughout the analysis, I've answered various business questions based on the data, including:**

1. How many sales transactions are there in our dataset?
2. What is the date range covered by the dataset?
3. Are there any outliers in car prices or gross profits? If so, how many are there and what are their values?
4. How many unique car models are present in the data?
5. What is the revenue comparison between the two car models (all-time)?
6. Which car model has the highest total gross profit (all-time)?
7. What is the annual total revenue generated by each car model in 2016 and 2017?
8. Which car model is the best performer in terms of revenue (all-time)?
9. How does the performance vary among different car versions within the best-performing model?
10. How does the performance vary among different car versions within the worst-performing model?
11. Which car version has the highest and lowest all-time sales?
12. Which car version generates the highest total gross profit (all-time)?
13. What is the most popular car version among buyers?
14. How many countries are there in the dataset?
15. What is the total gross profit and sales for each country?
16. Which country had the highest demand for Tesla cars?
17. How does the average gross profit vary across different car versions for each country?
18. What is the most sold car model in each country?
19. What is the most sold car version in each country?
20. What is the most expensive car version in each country?
21. What is the highest car version price in each country?
22. How many car models were sold in the US in 2016 and 2017?
23. Is there a difference in sales between the first and second quarter of 2016 and 2017?
24. Is there a difference in sales between car models in the first and second quarter of 2016 and 2017?
25. What is the total revenue for each month (all-time)?
26. What is the monthly gross profit for each month (all-time)?
27. What is the total revenue for each month in 2016 and 2017?
28. How many unique car models were sold in Australia?
29. What was the total gross profit generated from all sales in Germany?
30. What is the average revenue of all cars sold in the US?
31. What is the total gross profit generated from each car version in the US?
32. Which car model had the highest price in the US?
33. How many sales transactions were made for each car version in the US?
34. Which car model had the highest revenue in Australia?
35. What was the total sales generated from all sales in Australia?
36. How many purchase types are there in the dataset?
37. Which purchase type is most commonly used by people?
38. What is the distribution of purchase types for each car model?
39. What is the total sales for each purchase type?
40. What is the average gross profit for each purchase type?
41. Which country had the highest number of cash purchases?
42. Which country had the highest number of deposit purchases?
43. Which purchase type is most commonly used in each country?
44. What was the most common purchase type for 'Model S' in Germany?
45. How does the average price vary for different purchase types in the US?
46. Calculate the average car price and gross profit.
47. Calculate the standard deviation of price and gross profits.
48. Is there a correlation between car price and gross profit? If so, what is the correlation coefficient?



### **Let's do outliers detection and see if there's any outliers in the car prices or gross profits**

In [49]:
USE TeslaSales


SELECT 
  MAX(Revenue) AS max_price,
  MIN(Revenue) AS min_price,
  MAX(Gross_Profit) AS max_profit,
  MIN(Gross_Profit) AS min_profit
FROM CarSales;

(1 row affected)

Total execution time: 00:00:00.047

max_price min_price max_profit min_profit 88700.00 70700.00 54371.511 7178.568

There are no outliers since the numbers look normal.

### **determining the number of unique car models in the data.**

In [50]:
SELECT 
	COUNT(DISTINCT Model) AS no_of_models
FROM CarSales;

(1 row affected)

Total execution time: 00:00:00.112

no_of_models 2

There are indeed only the 2 models

### **Let’s see if there is a difference in sales between these two car models (all time).**

In [51]:
SELECT 
	Model, 
	SUM(Revenue) AS Total_Sales
FROM CarSales
GROUP BY Model
ORDER BY Total_Sales DESC;	

(2 rows affected)

Total execution time: 00:00:00.097

Model Total_Sales Model X 3555430000.00 Model S 2885637500.00

'Model X' generated $669,792,500.00 more revenue than 'Model S'. 

This difference in revenue is indeed significant and indicates that 'Model X' outperformed 'Model S' in terms of revenue generation during that specific 2-year timeframe.

### **Which model has the highest total gross profit (all time)?**

In [52]:
SELECT 
	Model, 
	SUM(Gross_Profit) AS Total_Profit
FROM CarSales
GROUP BY Model
ORDER BY Total_Profit DESC;

(2 rows affected)

Total execution time: 00:00:00.104

Model Total_Profit Model X 1039560332.872 Model S 838799261.395

'Model X' has the highest total profit with $1,039,560,332.87.

### **Let's analyze the annual total revenue generated by each of these car models in the year 2016**

In [53]:
SELECT 
	Model, 
	SUM(Revenue) AS Total_Sales
FROM CarSales
WHERE Period BETWEEN '2016-01-01' AND '2016-12-01'
GROUP BY Model
ORDER BY Total_Sales DESC;

(2 rows affected)

Total execution time: 00:00:00.130

Model Total_Sales Model X 1809338000.00 Model S 1443879100.00

"Model X" generated $365,458,900.00 more revenue than "Model S" in 2016.

### **Let's examine the total revenue generated by each of these car models in the year 2017.**

In [54]:
SELECT 
	Model, 
	SUM(Revenue) AS Total_Sales
FROM CarSales
WHERE Period BETWEEN '2017-01-01' AND '2017-12-01'
GROUP BY Model
ORDER BY Total_Sales DESC;

(2 rows affected)

Total execution time: 00:00:00.069

Model Total_Sales Model X 1746092000.00 Model S 1441758400.00

"Model X" generated $304,333,600.00 more revenue than "Model S" in 2017. <span style="color: var(--vscode-foreground);">So now we know our top performers and worst performers in this date range.&nbsp;</span>

### **What is our best performing model (all time)?**

In [55]:
SELECT 
	Model, 
	SUM(Revenue) AS Total_Sales
FROM CarSales
GROUP BY Model
ORDER BY Total_Sales DESC;

(2 rows affected)

Total execution time: 00:00:00.103

Model Total_Sales Model X 3555430000.00 Model S 2885637500.00

Our best performing model is "Model X" with total\_sales = $3,555,430,000.00

### **Let's analyze the performance of each car version within the best-performing model.**

In [56]:
SELECT Model, Version, SUM(Revenue) AS total_sales
FROM CarSales
WHERE Model = 'Model X'
GROUP BY Model, Version
ORDER BY total_sales DESC;

(2 rows affected)

Total execution time: 00:00:00.238

Model Version total_sales Model X P90D 2490470000.00 Model X 90D 1064960000.00

"P90D" car version generated $2,490,470,000.00 in revenue, while "90D" generated $1,064,960,000.00.

### **Let's see the performance of each car version within the worst-performing model.**

In [57]:
SELECT Model, Version, SUM(Revenue) AS total_sales 
FROM CarSales
WHERE Model = 'Model S'
GROUP BY Model, Version
ORDER BY total_sales DESC;

(4 rows affected)

Total execution time: 00:00:00.101

Model Version total_sales Model S 75 RWD 1165984400.00 Model S 60D AWD 670174400.00 Model S 75D AWD 655562000.00 Model S 90D AWD 393916700.00

It appears that the "75 RWD" is the top-selling car version in "Model S", while "90D AWD" is the least popular.

### **Which car version with the highest and lowest all-time sales.**

In [58]:
SELECT Version, SUM(Revenue) AS total_sales
FROM CarSales
GROUP BY Version
ORDER BY SUM(Revenue) DESC;

(6 rows affected)

Total execution time: 00:00:00.099

Version total_sales P90D 2490470000.00 75 RWD 1165984400.00 90D 1064960000.00 60D AWD 670174400.00 75D AWD 655562000.00 90D AWD 393916700.00

"P90D" has the highest sales with $2,490,470,000.00, while "90D AWD" has the lowest with $393,916,700.00.

### **Which car version is generating the highest total gross profit (all time)?**

In [59]:
SELECT 
	TOP 1 Version, 
		SUM(Gross_Profit) AS Total_Profit
FROM CarSales
GROUP BY Version
ORDER BY Total_Profit DESC;

(1 row affected)

Total execution time: 00:00:00.098

Version Total_Profit P90D 744668952.194

"P90D" generates the highest total gross profit among versions.

### **Which car version is the most popular among buyers?**

In [60]:
SELECT 
	Version, 
	COUNT(*) AS no_of_versions 
FROM CarSales
GROUP BY Version 
ORDER BY no_of_versions DESC;

(6 rows affected)

Total execution time: 00:00:00.098

Version no_of_versions P90D 33655 75 RWD 16492 90D 13312 60D AWD 9032 75D AWD 8660 90D AWD 4441

So we see that "P90D" is the most popular among buyers with 33,655 cars sold, while "90D AWD" is the least popular with 4,441 cars sold.

### **What is the total gross profit and sales for each country?**

In [61]:
SELECT 
	Country, 
	SUM(Gross_Profit) AS Total_Gross_Profit, 
	SUM(Revenue) AS Total_Sales
FROM CarSales
GROUP BY Country
ORDER BY Total_Gross_Profit DESC;

(3 rows affected)

Total execution time: 00:00:00.102

Country Total_Gross_Profit Total_Sales US 1523726361.090 4778070100.00 Germany 274277000.165 1284433000.00 Australia 80356233.012 378564400.00

### **Which country had the highest demand for Tesla cars?**

In [62]:
SELECT 
	Country, 
	COUNT(*) AS no_demand_cars
FROM CarSales
GROUP BY Country
ORDER BY no_demand_cars DESC;

(3 rows affected)

Total execution time: 00:00:00.070

Country no_demand_cars US 63469 Germany 16956 Australia 5167

"US" had the highest demand for Tesla cars with 63,469 cars, followed by "Germany" with 16,956 cars.

### **How does the average gross profit vary across different car versions for each country?**

In [63]:
SELECT 
	Country, 
	Version, 
	AVG(Gross_Profit) AS Avg_Gross_Profit
FROM CarSales
GROUP BY Country, Version
ORDER BY Country, Avg_Gross_Profit DESC;

(18 rows affected)

Total execution time: 00:00:00.124

Country Version Avg_Gross_Profit Australia 90D AWD 43921.662117 Australia 75D AWD 38953.293857 Australia 60D AWD 20335.792915 Australia 90D 17356.315207 Australia P90D 15540.389983 Australia 75 RWD 14121.527923 Germany 75D AWD 20586.564677 Germany 90D AWD 18414.889196 Germany P90D 16185.988954 Germany 90D 15446.391892 Germany 60D AWD 14668.898754 Germany 75 RWD 12987.580455 US 90D AWD 32712.510861 US 75D AWD 28001.642209 US P90D 24360.770655 US 90D 24141.851173 US 60D AWD 22968.265349 US 75 RWD 19237.203452

### **What is The most sold car model in each country?**

In [64]:
SELECT Country, Model
FROM (SELECT Country, Revenue, Model, ROW_NUMBER() OVER(PARTITION BY Country ORDER BY SUM(Revenue) DESC) AS RN
		FROM CarSales
		GROUP BY Country, Revenue, Model) as NewTable
WHERE RN = 1; 

(3 rows affected)

Total execution time: 00:00:00.162

Country Model Australia Model X Germany Model X US Model X

"Model X" is the most popular among countries

### **what is the most sold car version in each country?**

In [65]:
SELECT Country, Version
FROM (
  SELECT Country, Version, 
         ROW_NUMBER() OVER (PARTITION BY Country ORDER BY COUNT(*) DESC) AS RN
  FROM CarSales
  GROUP BY Country, Version
) AS sales_version
WHERE RN = 1;

(3 rows affected)

Total execution time: 00:00:00.104

Country Version Australia P90D Germany P90D US P90D

"P90D" is the most sold car version in each country

### **what is the most expensive car version in each country?**

In [66]:
SELECT Country, Version
FROM (
  SELECT Country, Version, 
         ROW_NUMBER() OVER (PARTITION BY Country ORDER BY Revenue DESC) AS RN
  FROM CarSales
  GROUP BY Country, Version, Revenue
) AS sales_version
WHERE RN = 1;

(3 rows affected)

Total execution time: 00:00:00.146

Country Version Australia 90D AWD Germany 90D AWD US 90D AWD

"90D AWD" is the most expensive car version in each country

### **what is the highest car version price in each country?**

In [67]:
SELECT Country, Revenue
FROM (
  SELECT Country, Revenue, 
         ROW_NUMBER() OVER (PARTITION BY Country ORDER BY Revenue DESC) AS RN
  FROM CarSales
  GROUP BY Country, Revenue
) AS sales_version
WHERE RN = 1; 

(3 rows affected)

Total execution time: 00:00:00.093

Country Revenue Australia 88700.00 Germany 88700.00 US 88700.00

The '90D AWD' car version reached the highest price of $88,700.00 individually in all three countries

### **How many car models were sold in the US in 2016?**

In [68]:
SELECT 
	Model, 
	COUNT(Model) AS us_no_models
FROM CarSales
WHERE (Country = 'US') AND (Period between '2016-01-01' AND '2016-12-01')              
GROUP BY Model;	

(2 rows affected)

Total execution time: 00:00:00.063

Model us_no_models Model X 17553 Model S 14592

\-- USA car sales in 2016:

\-- "Model X" was sold 17,553 times.

\-- "Model S" was sold 14,592 times.

### **How many car models were sold in the US in 2017?**

In [69]:
SELECT 
	Model, 
	COUNT(Model) AS us_no_models
FROM CarSales
WHERE (Country = 'US') AND (Period BETWEEN '2017-01-01' AND '2017-12-01')              
GROUP BY Model;	

(2 rows affected)

Total execution time: 00:00:00.151

Model us_no_models Model X 17238 Model S 14086

\-- USA car sales in 2017:

\-- "Model X" was sold 17,238 times.

\-- "Model S" was sold 14,086 times.

### **IS there a difference in sales between the first and second quarter of the year 2016 and 2017.?**

In [70]:
SELECT 
	SUM(Revenue) AS quarter_Revenue
FROM CarSales
WHERE Period between '2016-01-01' AND '2016-06-01';

(1 row affected)

Total execution time: 00:00:00.023

quarter_Revenue 1620321400.00

### **What is the total revenue for each month in 2016?**

In [71]:
SELECT month_name, sum(Revenue) as month_revenue
FROM CarSales
WHERE Period between '2017-01-01' AND '2017-12-01'
GROUP BY month_name 
ORDER BY month_revenue DESC;

(12 rows affected)

Total execution time: 00:00:00.074

month_name month_revenue July 282845500.00 June 281608700.00 April 277038700.00 August 274928300.00 March 272774400.00 May 268954300.00 February 267379500.00 September 262681600.00 December 257355700.00 January 249610800.00 November 248669000.00 October 244003900.00

"July" has the highest revenue with $282,845,500.00 and "October" is the lowest revenue with $244,003,900.00

### **What is the average Revenue of all cars sold in the US?**

In [72]:
SELECT 
	Country, 
	ROUND(AVG(Revenue),3) AS US_avg_Price
FROM CarSales 
WHERE Country = 'US'
Group BY Country;

(1 row affected)

Total execution time: 00:00:00.068

Country US_avg_Price US 75281.950000

The "US" has an average Revenue of all cars sold with $75281.950

### **What is the total gross profit generated from each car version in the US?**

In [73]:
SELECT 
	Version, 
	SUM(Gross_Profit) AS US_totalprofit
FROM CarSales
WHERE Country = 'US'
GROUP BY Version;

(6 rows affected)

Total execution time: 00:00:00.083

Version US_totalprofit P90D 600103224.326 75D AWD 180274572.544 60D AWD 170906862.469 90D AWD 103273396.790 75 RWD 223959522.592 90D 245208782.369

### **Which car model had the highest price in the US?**

In [74]:
SELECT 
	Model, 
	ROUND(SUM(Revenue), 2) AS US_total_price
FROM CarSales
WHERE Country = 'US'
GROUP BY Model
ORDER BY US_total_price DESC;

(2 rows affected)

Total execution time: 00:00:00.097

Model US_total_price Model X 2635476000.00 Model S 2142594100.00

"Model X" had the highest price in the US with $2,635,476,000.00

### **How many sales transactions were made for each car version in the US?**

In [75]:
SELECT 
	Version, 
	COUNT(*) AS no_sal_trans
FROM CarSales
WHERE Country = 'US'
GROUP BY Version;

(6 rows affected)

Total execution time: 00:00:00.100

Version no_sal_trans P90D 24634 75D AWD 6438 60D AWD 7441 90D AWD 3157 75 RWD 11642 90D 10157

### **Which car model had the highest revenue in Australia?**

In [76]:
SELECT 
	Model, 
	ROUND(SUM(Revenue), 2) AS AU_total_revenue
FROM CarSales
WHERE Country = 'Australia'
GROUP BY Model
ORDER BY AU_total_revenue DESC;	

(2 rows affected)

Total execution time: 00:00:00.046

Model AU_total_revenue Model X 205742000.00 Model S 172822400.00

"Model X" had the highest price in Australia with $205,742,000.00

### <span style="font-size: 16.38px;"><b>Which purchase type is most commonly used by people?</b></span>

In [77]:
SELECT 
	TOP 1 Purchase_type, 
	COUNT(*) AS no_of_purchases 
FROM CarSales
GROUP BY Purchase_type
ORDER BY no_of_purchases DESC;

(1 row affected)

Total execution time: 00:00:00.101

Purchase_type no_of_purchases Deposit 45976

"Deposit" is the most commonly purchase type with 45,976 number of purchases.

### **What is the distribution of purchase types for each car model?**

In [78]:
SELECT 
	Model, 
	Purchase_type, 
	COUNT(*) AS Frequency
FROM CarSales
GROUP BY Model, Purchase_type
ORDER BY Model, Frequency DESC;

(4 rows affected)

Total execution time: 00:00:00.196

Model Purchase_type Frequency Model S Deposit 20852 Model S Cash purchase 17773 Model X Deposit 25124 Model X Cash purchase 21843

### **What is the average gross profit for each purchase type?**

In [79]:
SELECT 
	Purchase_type, 
	AVG(Gross_Profit) AS AVG_Total_Sales
FROM CarSales
GROUP BY Purchase_type
ORDER BY AVG_Total_Sales DESC;

(2 rows affected)

Total execution time: 00:00:00.146

Purchase_type AVG_Total_Sales Deposit 22095.055110 Cash purchase 21771.944176

### **Which country had the highest number of cash purchases?**

In [80]:
SELECT 
	TOP 1 Country, 
	COUNT(*) AS cash_Purchase
FROM CarSales
WHERE Purchase_type = 'Cash purchase'
GROUP BY Country
ORDER BY cash_Purchase DESC;

(1 row affected)

Total execution time: 00:00:00.068

Country cash_Purchase US 28788

The United States (US) recorded the highest number of cash purchases, with a total of 28,788 transactions.

### **Which country had the highest number of Deposit purchases?**

In [81]:
SELECT 
	TOP 1 Country, 
	COUNT(*) AS depo_Purchase
FROM CarSales
WHERE Purchase_type = 'Deposit'
GROUP BY Country
ORDER BY depo_Purchase DESC;

(1 row affected)

Total execution time: 00:00:00.166

Country depo_Purchase US 34681

we also see that the United States (US) recorded the highest number of Deposit with 34,681 purchases transactions.

### **Which purchase type is most commonly used in each country?**

In [82]:
SELECT Country, Purchase_type
FROM (SELECT Country, Purchase_type, ROW_NUMBER() OVER(PARTITION BY Country ORDER BY count(*) DESC) AS RN
		FROM CarSales
		GROUP BY Country, Purchase_type) as NewTable
WHERE RN = 1;

(3 rows affected)

Total execution time: 00:00:00.108

Country Purchase_type Australia Deposit Germany Deposit US Deposit

"Deposit" is the most commonly used purchase type in each country.

### **How does the average price vary for different purchase types in the US?**

In [83]:
SELECT 
	Purchase_type, 
	ROUND(AVG(Revenue), 3) AS avg_sales
FROM CarSales
WHERE Country = 'US'
GROUP BY Purchase_type;	

(2 rows affected)

Total execution time: 00:00:00.113

Purchase_type avg_sales Cash purchase 75226.455000 Deposit 75328.015000

the "Cash purchase" method showed an average price of $75,226.455, while the "Deposit" method had a slightly higher average price of $75,328.015.

### **Calculating the average car price and gross profit**

### **Calculating the standard deviation of price and gross profits**

In [84]:
SELECT
  AVG(Revenue) AS Price_Mean,
  STDEV(Revenue) AS Price_StdDev,
  AVG(Gross_Profit) AS Profit_Mean,
  STDEV(Gross_Profit) AS Profit_StdDev
FROM CarSales;

(1 row affected)

Total execution time: 00:00:00.117

Price_Mean Price_StdDev Profit_Mean Profit_StdDev 75253.148658 4199.494953581349 21945.504185 10328.412364791378

### **Is there a correlation between car price and gross profit?**

In [85]:
SELECT 
  (SUM(Revenue * Gross_Profit) - COUNT(*) * AVG(Revenue) * AVG(Gross_Profit)) /
  (COUNT(*) * STDEV(Revenue) * STDEV(Gross_Profit)) AS CorrelationCoefficient
FROM CarSales;

(1 row affected)

Total execution time: 00:00:00.114

CorrelationCoefficient 0.20384217667006208

Yes, there is a correlation between car price and gross profit.

The correlation coefficient of 0.2039 between car price and gross profit is positive and close to zero, but not very strong. 

 This suggests that there is a slight positive linear relationship between car price and gross profit, meaning that as car prices increase, 

 there is a tendency for gross profit to also increase, but the relationship is not very strong.

# **Dashboard**

## **For the Dashboard, check [Tesla_Car_Sales_Dashboard](https://www.novypro.com/project/tesla-car-sales) link**


![photo_5942650041216912698_y](https://github.com/MohamedMohsen01/SQL-Projects/assets/109850173/cdd8f181-e8cf-44b3-a63c-64bef7db97f1)


![photo_5942650041216912700_y](https://github.com/MohamedMohsen01/SQL-Projects/assets/109850173/275977ed-a989-435a-bf85-eb451e2cd0cc)

# **Results and Insights**

- 'Model X' outperformed 'Model S' in terms of revenue, with a difference of $669,8 million during the entire timeframe.

- 'Model X' generated the highest total gross profit, amounting to $1,039,560,332.

- In 2016, 'Model X' achieved $1,809,338,000.00 in total sales, while 'Model S' had $1,443,879,100.00.  In 2017, - - 'Model X' reached $1,746,092,000.00 in total sales, and 'Model S' had $1,441,758,400.

- 'Model X' emerged as the best-performing model with total sales of $3,555,430,000.

- Among the car versions in 'Model X', the 'P90D' version achieved the highest revenue of $2,490,470,000.00, while the '90D' version generated $1,064,960,000.

- Within 'Model S', the '75 RWD' version was the top-selling car version, while the '90D AWD' version had the least sales.

- The 'P90D' car version had the highest sales with $2,490,470,000.00, while the '90D AWD' car version had the lowest sales with $393,916,700.00.

- The 'P90D' car version generated the highest total gross profit among all versions.

- The 'P90D' car version was the most popular among buyers, with 33,655 cars sold, while the '90D AWD' version had 4,441 sales.

- The US had the highest demand for Tesla cars, with 63,469 cars sold, followed by Germany with 16,956 cars.

- Among different car versions in Australia, the '90D AWD' version had the highest average gross profit, while in Germany, the '75D AWD' version had the highest average gross profit. In the US, the '90D AWD' version had the highest average gross profit.

- The 'Model X' was the most popular car model in all countries.

- The 'P90D' car version was the most sold version in each country.

- The '90D AWD' car version was the most expensive version in each country.

- The highest car price in all three countries was $88,700.00 for the '90D AWD' car version.

- In 2016, 'Model X' was sold 17,553 times, and 'Model S' was sold 14,592 times in the USA. In 2017, 'Model X' was sold 17,238 times, and 'Model S' was sold 14,086 times.

- The data showed a difference in sales between the first and second quarter of 2016. However, there was no significant difference between the first and second quarter of 2017.

- There was a difference in sales between car models in the first and second quarter of 2016. However, the difference was not substantial in the first and second quarter of 2017.

- July had the highest revenue with $564,770,000.00, while January had the lowest revenue.

- Gross profits followed a similar trend to revenue, with July having the highest monthly gross profit.

- April had the highest revenue in 2016 with $283,424,700.00, and July had the highest revenue in 2017 with $282,845,500.00. October had the lowest revenue in 2017 with $244,003,900.00.

- A total of two unique car models were sold in Australia.

- Germany generated a total gross profit of $378,564,400.00 from all sales.

- The average revenue of all cars sold in the US was $75,281.950.

- The '90D AWD' car version had the highest total gross profit in the US.

- 'Model X' had the highest price in the US with $2,635,476,000.00.

- The '90D AWD' car version had the highest number of sales transactions in the US.

- 'Model X' had the highest revenue in Australia with $205,742,000.00.

- The "Deposit" purchase type was the most commonly used, with 45,976 transactions.

- The "Deposit" purchase type showed an average gross profit of $22,095.055, while "Cash Purchase" had an average gross profit of $21,771.944.

- The US had the highest number of cash purchases, with 30,628 transactions.

- The US had the highest number of deposit purchases, with 30,841 transactions.

# **Conclusion**

Based on the comprehensive exploratory data analysis conducted on the Tesla car sales data, several valuable insights have been obtained. 

- 'Model X' emerged as the top-performing car model with the highest total revenue and gross profit, while the 'P90D' car version stood out as the most popular and highest-grossing version.

- The US was the most significant market for Tesla cars, contributing the highest number of sales and revenue. It also displayed a strong preference for the "Deposit" purchase type. The '90D AWD' car version demonstrated consistent performance, achieving the highest average gross profit in multiple regions.

- Understanding the performance trends, key selling months, and purchase preferences will assist in devising effective sales strategies and driving revenue growth. It is recommended to focus marketing efforts on the 'Model X' and 'P90D' version, particularly in regions with high demand. Additionally, exploring the reasons behind the popularity of the "Deposit" purchase type may provide valuable insights into consumer behavior.

- It is crucial to continuously monitor and update the sales strategy based on changing market dynamics to stay competitive and maximize profitability. The correlation between car price and gross profit emphasizes the importance of pricing decisions in achieving higher profitability.

# **Recommendations**


- Focus on promoting and marketing the 'Model X' as it has been the top-performing car model in terms of total revenue and gross profit. Highlight its unique features and advantages to attract more customers.
- Consider offering special promotions or incentives for the 'P90D' car version, as it has proven to be the most popular and highest-grossing version. This can further boost its sales and profitability.

- Concentrate marketing efforts on regions with high demand for Tesla cars, particularly the United States and Germany. These regions have shown the highest sales and revenue, making them crucial markets for continued growth.
- Analyze the factors contributing to the success in these regions and tailor marketing strategies to address the specific preferences and needs of the customers there.

- Leverage the knowledge of high-performing months, such as July, to plan targeted marketing campaigns and promotions during peak demand periods.
- Analyze the reasons behind the difference in sales between the first and second quarter of 2016 and utilize the insights to optimize sales tactics during similar periods in the future.

- Continue to monitor the correlation between car price and gross profit. Consider conducting price elasticity analysis to understand how pricing changes may impact profitability.
- Explore dynamic pricing strategies that take into account market demand and customer preferences to optimize revenue and gross profit.

- Investigate the reasons behind the popularity of the "Deposit" purchase type and identify the factors driving this preference. Use this information to tailor marketing messages and incentives to attract more customers to this purchase option.


- Establish a system for regular data monitoring and analysis to keep track of changing market dynamics and customer preferences.
- Use insights from ongoing analysis to continuously refine sales strategies and improve overall performance.


- Conduct a competitive analysis to understand the strengths and weaknesses of competitors in the electric vehicle market. Use this information to identify opportunities for differentiation and innovation.

- Collect customer feedback and conduct surveys to gauge customer satisfaction levels and identify areas for improvement in products and services.
- Address any issues or concerns raised by customers promptly to ensure a positive customer experience and enhance brand loyalty.
